### PR TITLE
Adding Cause#untraced

### PR DESCRIPTION
**Pull request by shankarshastri**
Added Cause#untraced
https://github.com/zio/zio/issues/1317

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZManaged.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ import zio.clock.Clock
 import zio.duration.Duration
 
 /**
- * A `Reservation[-R, +E, +A]` encapsulates resource aquisition and disposal
+ * A `Reservation[-R, +E, +A]` encapsulates resource acquisition and disposal
```
Comment (kaishh) [inline core/shared/src/main/scala/zio/ZManaged.scala]: nice catch!

**Comment by kaishh** [comment]
@shankarshastri The lint step on CI is failing, you can fix this by running `sbt fmt` and pushing the changes

**Comment by shankarshastri** [comment]
@kaishh , Thanks. fixed and pushed the same.

File: core-tests/shared/src/test/scala/zio/ExitSpec.scala [hunk 2]
```diff
@@ -11,6 +11,7 @@ class ExitSpec extends Specification with ScalaCheck {
       `Cause#died` and `Cause#stripFailures` are consistent $e1
       `Cause.equals` is symmetric $e2
       `Cause.equals` and `Cause.hashCode` satisfy the contract $e3
+      `Cause#untraced` satisfy the contract $e8
```
Comment (kaishh) [inline core-tests/shared/src/test/scala/zio/ExitSpec.scala]: You may reword this as as `Cause#untraced removes all traces`, note that "the contract" on the line above refers to the contracts of .equals() and .hashCode() as defined by Java Specification; there's no such widely known contract attached to `untraced`, so the current wording may seem somewhat off

Comment (shankarshastri) [reply core-tests/shared/src/test/scala/zio/ExitSpec.scala]: Sure.

**Comment by kaishh** [comment]
@shankarshastri Thank you very much for contributing and congratulations on your first ZIO pull request!

giphy
